File: CSharp-Part-1/HW3.OperatorsAndExpressions/T07.PointInCircle/PointInCircle.cs
namespace T07.PointInCircle
{
    using System;
    using System.Linq;

    /// <summary>
    /// 7. Write an expression that checks if given point (x, y) is inside a circle K({0, 0}, 2).
    /// </summary>
    public class PointInCircle
    {
        private static void Main()
        {
            int x = ReadAndValidateInput("Please enter value for X = ");
            int y = ReadAndValidateInput("Please enter value for Y = ");
            const int K = 2;

            int circle = (x * x) + (y * y);

            if (circle <= K * K)
            {
                Console.WriteLine("Points (x = {0}, y = {1}) are in circle K(O, 2)", x, y);
            }
            else
            {
                Console.WriteLine("Points (x = {0}, y = {1}) are outside of circle K(O, 2)", x, y);
            }
        }

        /// <summary>
        /// Read and validate user input as integer
        /// </summary>
        /// <param name="msg">input message</param>
        /// <returns>int result</returns>
        private static int ReadAndValidateInput(string msg)
        {
            int number = 0;

            Console.Write(msg);

            bool isValidIntegerWidth = int.TryParse(Console.ReadLine(), out number);
            while (isValidIntegerWidth == false)
            {
                Console.Write("Invalid number! {0}", msg);
                isValidIntegerWidth = int.TryParse(Console.ReadLine(), out number);
            }

            return number;
        }
    }
}

File: JavaScript.Funamentals/4.Operators-and-Expressions/scripts/point-in-circle.js
(function () {
    "use strict";

    var points;
    var i;
    var x;
    var y;
    var radius = 5;

    points = [
        {x: 0, y: 1},
        {x: -5, y: 0},
        {x: -4, y: 5},
        {x: 1.5, y: -3},
        {x: -4, y: -3.5},
        {x: 100, y: -30},
        {x: 0, y: 0},
        {x: 0.2, y: -0.8},
        {x: 0.9, y: -4.93},
        {x: 2, y: 2.655}
    ];

    function inCircle(x,y){
        var circle = (x * x) + (y * y);
        var diameter = radius * radius;

        return circle <= diameter;
    }

    for(i = 0; i < points.length; i++){
        x = points[i].x;
        y = points[i].y;

        if(inCircle(x,y)){
            console.log("True: " + "Points " + x + " and " + y + " are in circle K(0, " + radius + ")");
        } else {
            console.log("False: " + "Points " + x + " and " + y + " are not in circle K(0, " + radius + ")")
        }
    }
}());